sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/joinTypes.scala

```diff
@@ -47,4 +48,9 @@ case object RightOuter extends JoinType
 
 case object FullOuter extends JoinType
 
-case object LeftSemi extends JoinType
+abstract class LeftSemiJoin extends JoinType
```

Comment:
Why have a abstract class at here? Also, looks like `LeftSemiJoin` is not a good name since `LeftAnti` extends it and `LeftAnti` is a different concept.